roda-core/roda-core/src/main/java/org/roda/core/plugins/base/synchronization/instanceIdentifier/InstanceIdentifierPreservationAgentPlugin.java

```diff
@@ -136,14 +140,27 @@ private void modifyInstanceId(ModelService model, IndexService index, Job cached
 
     Report reportItem = PluginHelper.initPluginReportItem(this, cachedJob.getId(), Job.class);
     PluginHelper.updatePartialJobReport(this, model, reportItem, false, cachedJob);
+    DistributedInstances distributedInstances = null;
+    try {
+      distributedInstances = RodaCoreFactory.getModelService().listDistributedInstances();
+    } catch (RequestNotValidException | GenericException | AuthorizationDeniedException | IOException e) {
+      LOGGER.error("Could not list distributed instances");
+    }
+    List<String> distributedInstanceIds = distributedInstances.getObjects().stream()
+      .map(distributedInstance -> distributedInstance.getId()).collect(Collectors.toList());
```

Comment:
If you have an exception while retrieving distributed instances, you will get NullPointerExceptions when handling the distributedInstances object